### ISSUE TITLE
`u-off-screen` does not disable itself when descendant elements are focused

### ISSUE THREAD

**Issue by jmuzina** [canonical/vanilla-framework#5734]
`u-off-screen` does not un-hide the element it is applied to when focus is moved to a descendant of the `u-off-screen` element. 

I believe we should consider un-hiding `u-off-screen` elements when focus is moved to within them. This would expand the interactivity of [SkipLink](https://vanillaframework.io/docs/patterns/links#skip-link), which already has this behavior, to more complex cases where multiple interactive elements are hidden and revealed as a group.

 Consider a visually hidden Form (maybe it is an accessible alternative to a more complex set of controls that are not possible to use with a keyboard). You would apply `u-off-screen` to the form element, and expect that tabbing into one of the child form input controls would cause the entire form to become visible. See [Codepen](https://codepen.io/jmuzina/pen/EayZOBR)

However, this is not the case currently because we only cancel our absolute positioning when `:focus` is applied to the off-screen element itself. This means that focus seems to disappear when it moves to the hidden element.

https://github.com/canonical/vanilla-framework/blob/ad94436af9f0635624e2111045a08492bfd9b466/scss/_base_placeholders.scss#L135-L144

**Comment by dgtlntv** [comment]
Design systems contribution meeting:
* We should aim to make things keyboard intractable in the first place.
* If that is not possible due to some reasons then this seems like a good alternative.
* We should have better documentation for when using this pattern is appropriate before it gets accepted into the design system.
